### ISSUE TITLE
Including a '$' in Text throws off TextOverflow.ellipsis

### ISSUE THREAD

**Issue by CarlosZ** [flutter/flutter#45308]
## Steps to Reproduce

Consider the following snippet:

```dart
class MyTest extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SizedBox(
        width: 50,
        child: Container(
          color: Colors.red,
          child: Text(
            '* \$some long test',
            overflow: TextOverflow.ellipsis,
          ),
        ),
      ),
    );
  }
}
```

The above app renders like this:
<img width="273" alt="Screen Shot 2019-11-20 at 5 31 41 PM" src="https://user-images.githubusercontent.com/289158/69292765-c596b680-0bbb-11ea-80db-f1e4cc165527.png">

However if the Text is changed to `'* some long test'` (notice the lack of \$) then the ellipsizing works as expected:

<img width="273" alt="Screen Shot 2019-11-20 at 5 34 19 PM" src="https://user-images.githubusercontent.com/289158/69292861-055d9e00-0bbc-11ea-8bc5-6f2290db291e.png">

I would have expected that `'* \$some long test'` would ellipsize to something like `* $so...`

**Target Platform:** macOS
**Target OS version/browser:** Mojave
**Devices:** physical

## Logs

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel unknown, v1.12.3-pre.38, on Mac OS X 10.14.6 18G87, locale en-US)
[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/setup/#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, set ANDROID_HOME to that location.
      You may also want to add it to your PATH environment variable.

[!] Xcode - develop for iOS and macOS (Xcode 10.1)
    ✗ Flutter requires a minimum Xcode version of 10.2.0.
      Download the latest version or update via the Mac App Store.
[!] Android Studio (not installed)
[✓] IntelliJ IDEA Community Edition (version 2019.2.4)
[✓] Connected device (1 available)
```

**Comment by jason-simmons** [comment]
This happens because the current implementation of text layout breaks lines before applying the ellipsis.

The `$some` word exceeds the width constraint and is wrapped to the second line.  The only text that fits on the first line is the `*`, so the layout library renders that text plus the ellipsis.

**Comment by CarlosZ** [comment]
@jason-simmons I would have expected that when ellipsizing no line breaking would take place (or at least it would be controlled by setting `maxLines`). For example in HTML using the `text-overflow` CSS property looks at the width of the entire content and applies the ellipsis to the entire content without trying to break it into lines.

**Comment by krisgiesing** [comment]
@jason-simmons if what you say is true, then why doesn't the 2nd example also render as '* ...'? The 2nd word in that case is ellipsized in the middle.

**Comment by jason-simmons** [comment]
In the second example the word `some` (without the `$` character) fits on the first line.  Ellipsizing starts with the text that fits on the line and then truncates characters until there is enough space for the ellipsis.

**Comment by krisgiesing** [comment]
If I replace "some" in that test with "somereallyreally", the text still truncates in the middle of the word.

The `$` seems to be significant somehow.

**Comment by markusaksli-nc** [comment]
Closing since either this was never a valid issue or is fixed on the latest master `1.22.0-2.0.pre.78`.
<details>
 <summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Sample',
      home: Scaffold(
        appBar: AppBar(
          title: Text('Sample'),
        ),
        body: Center(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              SizedBox(
                width: 60,
                child: Text(
                  '* \$some long test',
                  overflow: TextOverflow.ellipsis,
                ),
              ),
              SizedBox(
                width: 60,
                child: Text(
                  '* 1some long test',
                  overflow: TextOverflow.ellipsis,
                ),
              ),
              SizedBox(
                width: 60,
                child: Text(
                  '* some long test',
                  overflow: TextOverflow.ellipsis,
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
```
</details>
<details>
 <summary>flutter doctor -v</summary>

```
[√] Flutter (Channel master, 1.22.0-2.0.pre.78, on Microsoft Windows [Version 10.0.19041.450], locale en-US)
    • Flutter version 1.22.0-2.0.pre.78 at C:\Development\flutter_master
    • Framework revision d0596582fe (2 hours ago), 2020-08-25 23:26:06 -0700
    • Engine revision b1d9f863db
    • Dart version 2.10.0 (build 2.10.0-56.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at C:\Users\marku\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.1
    • Java binary at: C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\193.6626763\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.6.5)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.6.30320.27
    • Windows 10 SDK version 10.0.18362.0

[√] Android Studio (version 4.0)
    • Android Studio at C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\193.6626763
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] Connected device (5 available)
    • SM G950F (mobile)           • ce12171c51cc001c03 • android-arm64  • Android 9 (API 28)
    • sdk gphone x86 arm (mobile) • emulator-5554      • android-x86    • Android 11 (API 30) (emulator)
    • Windows (desktop)           • windows            • windows-x64    • Microsoft Windows [Version 10.0.19041.450]
    • Web Server (web)            • web-server         • web-javascript • Flutter Tools
    • Chrome (web)                • chrome             • web-javascript • Google Chrome 84.0.4147.135

• No issues found!
```
</details>
